clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/api/KubernetesApiAdaptor.groovy

```diff
@@ -33,6 +40,29 @@ class KubernetesApiAdaptor {
     this.client = client
   }
 
+  /*
+   * Exponential backoff strategy for waiting on changes to replication controllers
+   */
+  Boolean blockUntilReplicationControllerConsistent(ReplicationController desired) {
+    def current = getReplicationController(desired.metadata.namespace, desired.metadata.name)
+
+    def wait = RETRY_INITIAL_WAIT_MILLIS
+    def attempts = 0
+    while (current.status.observedGeneration < desired.status.observedGeneration) {
+      attempts += 1
+      if (attempts > RETRY_COUNT) {
+        return false
+      }
+      sleep(wait)
```

Comment:
nit: put the wait assignment next to this statement (group the same variable usage)